src/endpoints/sources.js

```diff
@@ -305,39 +343,72 @@ function sendSubresource(subresourceExtractor) {
-     *
-     *    client (browser, CLI, etc) ⟷ us (nextstrain.org) ⟷ upstream source
-     */
-    return await proxyResponseBodyFromUpstream(req, res, new Request(subresourceUrl, {
-      headers: {
-        Accept: subresource.accept,
-
-        /* Normalize Accept-Encoding to gzip or nothing, as realistically we
-         * only expect upstreams to support gzip and normalizing reduces the
-         * cache burden of varying on many equivalent but effectively the same
-         * client Accept-Encoding values.
-         */
-        "Accept-Encoding": req.acceptsEncodings("gzip")
-          ? "gzip"
-          : "identity",
-      },
-
-      /* Use "no-cache" mode to always revalidate with the upstream but avoid
-       * re-transferring the content if we have a cached copy that matches
-       * upstream.
-       */
-      cache: "no-cache",
 
-      /* Disable automatic body decompression.  Instead, we pass through the
-       * Accept-Encoding of the client to the upstream and return the
-       * upstream Content-Encoding and body directly.  This avoids needlessly
-       * decompressing and recompressing ourselves.
+/**
+ * Proxy the data through us:
+ *
+ *    client (browser, CLI, etc) ⟷ us (nextstrain.org) ⟷ upstream source
+ *
+ * @param {express.request} req - Express-style request instance
+ * @param {express.response} res - Express-style response instance
+ * @param {String} url - Resource URL
+ * @param {String} accept - HTTP Accept header value
+ * @returns {expressEndpointAsync}
+ */
+async function proxyFromUpstream(req, res, url, accept) {
+  return await proxyResponseBodyFromUpstream(req, res, new Request(url, {
+    headers: {
+      Accept: accept,
+
+      /* Normalize Accept-Encoding to gzip or nothing, as realistically we
+       * only expect upstreams to support gzip and normalizing reduces the
+       * cache burden of varying on many equivalent but effectively the same
+       * client Accept-Encoding values.
        */
-      compress: false,
-    }));
+      "Accept-Encoding": req.acceptsEncodings("gzip")
+        ? "gzip"
+        : "identity",
+    },
+
+    /* Use "no-cache" mode to always revalidate with the upstream but avoid
+     * re-transferring the content if we have a cached copy that matches
+     * upstream.
+     */
+    cache: "no-cache",
+
+    /* Disable automatic body decompression.  Instead, we pass through the
+     * Accept-Encoding of the client to the upstream and return the
+     * upstream Content-Encoding and body directly.  This avoids needlessly
+     * decompressing and recompressing ourselves.
+     */
+    compress: false,
+  }));
+}
+
+
+/**
+ * Generate an Express endpoint that receives a group overview
+ * determined by the request.
+ *
+ * @returns {expressEndpointAsync}
+ */
+function receiveGroupOverview() {
+  return async (req, res) => {
+    const groupSource = req.context.source;
+
+    authz.assertAuthorized(req.user, authz.actions.Write, groupSource);
```

Comment:
Hmm, after 8aa733ca29f41f5eab8b2edbf8a060d966dbd937 seems like this doesn't work anymore:

```sh
curl http://localhost:5000/groups/test-private/overview \
    --header "Authorization: Bearer $token" \
    --header 'Content-Type: text/markdown' \
    --upload-file overview.md
```

I get a 403 when logged in as my user, a member of `test-private/owners`. Gonna take a deeper look at this next week, much easier now with [a nice VSCode debugging setup](https://gist.github.com/victorlin/e9e1177aef9cdd0d14ddbc10c54434da).

Reply:
Did your token expire?

Reply:
No, I'm able to `GET` the file right before `PUT`. If I pass an invalid token (I assume this is equivalent to an expired token) I get `401 Unauthorized` which makes sense, but this is `403 Forbidden`.

Reply:
This was fixed by 5fce14b7cbac318456f293cece839847797eab7e.